python/lsst/ts/MTMount/mtmount_csc.py

```diff
@@ -248,30 +247,22 @@ async def begin_enable(self, data):
                     f"The CSC was not allowed to command the mount: {e!r}"
                 )
 
-        self.enable_task.cancel()
         self.enable_task = asyncio.create_task(self.enable_devices())
-        await self.enable_task
-
-    async def begin_disable(self, data):
         try:
-            await super().begin_disable(data)
-            if self.has_control and self.connected:
-                self.disable_task.cancel()
-                self.disable_task = asyncio.create_task(self.disable_devices())
-                await self.disable_task
+            await self.enable_task
+        except Exception:
+            self.log.warning(
+                "Could not enable devices; disabling devices and giving up control."
+            )
+            self.disable_task = asyncio.create_task(self.disable_devices())
+            await self.disable_task
+            raise
 
-            try:
-                self.log.info("Give up command of the mount.")
-                await self.send_command(
-                    commands.AskForCommand(commander=enums.Source.NONE)
-                )
-                self.has_control = True
-            except Exception as e:
-                self.log.warning(
-                    f"The CSC was not able to give up command of the mount: {e!r}"
-                )
-        finally:
-            self.has_control = False
+    async def begin_disable(self, data):
+        await super().begin_disable(data)
+        self.disable_task.cancel()
+        self.disable_task = asyncio.create_task(self.disable_devices())
+        await self.disable_task
```

Comment:
The same as for enable_task shall apply here. Why not to replace create_task(t) && await with await t?

Reply:
Same answer.